R/api.R

```diff
@@ -16,108 +16,360 @@
   get_content(hn_api_response)
 }
 
-#' Retrieves the item corresponding to the specified id
+
+#' Checks whether max_items is correctly defined. If not throws an error
 #'
-#' @export
+#' @param max_items maximum number of ids to retrieve
+#
+.assert_max_items <- function(max_items) {
+  if (!is.null(max_items)) {
+    assert(is.numeric(max_items), "max_items must be a numeric type")
+    assert(max_items > 0,
+           "max_items must be greater than 0")
+  }
+}
+
+#' Checks whether ids are correctly defined. If not throws an error
+#'
+#' @param ids maximum number of ids to retrieve
+#
+.assert_ids <- function(ids) {
+    assert(all(is.numeric(ids)), "every id must be a numeric type")
+    assert(all(ids > 0),
+           "every id must be greater than 0")
+}
+
+
+#' Selects only limited number of ids
+#'
+#' @param ids_list list of ids to be trimed
+#'
+#' @param max_items maximum number of ids to retrieve
+#'
+#' @return list of ids
+#'
+.trim_ids_list <- function(ids_list, max_items) {
```

Comment:
Same as above but this time maybe let's move to `utils.R`? (If we do we need to remove the dot at the beginning of the function name)